quadriga/src/main/java/edu/asu/diging/quadriga/config/web/CitesphereTokenFilter.java

```diff
@@ -0,0 +1,64 @@
+package edu.asu.diging.quadriga.config.web;
+
+import java.io.IOException;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.http.HttpHeaders;
+import org.springframework.security.authentication.AuthenticationCredentialsNotFoundException;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.AbstractAuthenticationProcessingFilter;
+
+import edu.asu.diging.quadriga.core.model.citesphere.CitesphereToken;
+
+/**
+ * This class acts as a filter to check whether the request has a bearer
+ * authorization token in its header and authenticates the request.
+ * 
+ * @author poojakulkarni
```

Comment:
Since this is pretty much a copy of the class in Giles, there should be a least a link indicating that this class is based on someone else's code.